File: CS/EmployeeDataSource.cs
using System.Collections.Generic;

namespace Reporting_Create_Report_Parameter_with_Predefined_Dynamic_Values {
    public class Employee {
        public string Name { get; set; }
        public string Position { get; set; }
    }

    public class EmployeeDataSource {
        private List<Employee> employees = new List<Employee>() {
            new Employee() {
                Name = "Antonio Moreno",
                Position = "CEO"
            },
            new Employee() {
                Name = "Thomas Hardy",
                Position = "Sales Representative"
            },
            new Employee() {
                Name = "Christina Berglund",
                Position = "Order Administrator"
            },
            new Employee() {
                Name = "Frédérique Citeaux",
                Position = "Marketing Manager"
            },
            new Employee() {
                Name = "Hanna Moos",
                Position = "Software Developer"
            }
        };

        public IEnumerable<Employee> GetEmployeeList() {
            foreach (var employee in employees)
                yield return employee;
        }
    }
}

File: VB/EmployeeDataSource.vb
Imports System.Collections.Generic

Namespace Reporting_Create_Report_Parameter_with_Predefined_Dynamic_Values
	Public Class Employee
		Public Property Name() As String
		Public Property Position() As String
	End Class

	Public Class EmployeeDataSource
		Private employees As New List(Of Employee)() From {
			New Employee() With {
				.Name = "Antonio Moreno",
				.Position = "CEO"
			},
			New Employee() With {
				.Name = "Thomas Hardy",
				.Position = "Sales Representative"
			},
			New Employee() With {
				.Name = "Christina Berglund",
				.Position = "Order Administrator"
			},
			New Employee() With {
				.Name = "Frédérique Citeaux",
				.Position = "Marketing Manager"
			},
			New Employee() With {
				.Name = "Hanna Moos",
				.Position = "Software Developer"
			}
		}

		Public Iterator Function GetEmployeeList() As IEnumerable(Of Employee)
			For Each employee In employees
				Yield employee
			Next employee
		End Function
	End Class
End Namespace
